python/looker/rtl/user_session.py

```diff
@@ -61,48 +52,58 @@ def get_token(self):
             self.login()
         return self._token
 
-    def authenticate(self):
-         # TODO: still to be built based on request
-        return
+    def authenticate(self) -> Dict[str, str]:
+        token = self.get_token()
+        header = {}
+        if token and token.access_token:
+            header['Authorization'] = f'token {token.access_token}'
+        return header
 
     def login(self, user_id: Optional[str] = None) -> at.AuthToken:
         if not self.is_authenticated:
-            token = self.__login(user_id)
+            token = self._login(user_id)
             self._token = at.AuthToken(token)
-            return token
         return self._token
 
-    def __login(self, user_id: Optional[str]) -> tp.SDKSuccessResponse.value:
+    def _login(self, user_id: Optional[str]) -> tp.Response.value:
         if (user_id and self.user_id != user_id):
             # We are switching user ids
-            self.__logout()
+            self._logout()
 
-        self.__reset()
+        self._reset()
 
         # pylint: disable=too-many-function-args
-        result = self.__ok(self.transport.request(tp.HttpMethod.POST,
-                                                  '/login',
-                                                  body={
-                                                      'client_id': self.settings.client_id,
-                                                      'client_secret': self.settings.client_secret
-                                                  }))
+        result = self._ok(
+            self.transport.request(tp.HttpMethod.POST,
+                                   '/login',
+                                   body={
+                                       'client_id': self.settings.client_id,
+                                       'client_secret':
+                                       self.settings.client_secret
+                                   }))
         return result
 
     def logout(self) -> bool:
-        result = False
         if self.is_authenticated:
-            result = self.__logout()
-        return result
+            result = self._logout()
+        return bool(result)
 
-    def __logout(self):
+    def _logout(self) -> tp.Response.value:
         token = self._token
-        # TODO: this is still to be built based on authenticator
+        result = self._ok(
+            self.transport.request(
+                tp.HttpMethod.DELETE,
+                '/logout',
+                authenticator=(lambda: {
+                    'Authorization': f'token {token.access_token}'
+                }) if token else None))
```

Comment:
this is a little tricky to read ... can the `lambda` here return `None`? Looks like `_logout` is gated by an `is_authenticated` check so we should be sure of having a `token`. So think we can get rid of the ternary here. Also, I believe lambda's in python are more like arrow functions in js and so `self` references the enclosing instance (e.g. I think you can do ...`f'token {self._token.access_token}` and avoid L92)

Reply:
Nice catch, I'll change this.